src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -344,6 +344,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <PropertyGroup>
       <_IgnoreLinkerWarnings>false</_IgnoreLinkerWarnings>
       <_IgnoreLinkerWarnings Condition="'$(TargetOS)' == 'osx'">true</_IgnoreLinkerWarnings>
+      <StripFlag Condition="'$(TargetOS)' == 'osx' and '$(NativeLib)' == 'Shared'">-x</StripFlag> <!-- keep global symbols in dylib -->
```

Comment:
Do we have any global symbols in executables? I am wondering whether this should be set for dylibs and executables.

Reply:
Yes there are some special symbols left in executable, noticeably uuid for debuggers to recognize the symbol files (PR description has an example).